google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -302,17 +302,37 @@ static boolean isDefaultStream(String streamName) {
     return streamMatcher.find();
   }
 
-  private BigQueryWriteSettings getBigQueryWriteSettings(Builder builder) throws IOException {
+  static BigQueryWriteSettings getBigQueryWriteSettings(Builder builder) throws IOException {
+    BigQueryWriteSettings.Builder settingsBuilder = null;
     if (builder.client != null) {
-      return builder.client.getSettings();
+      settingsBuilder = builder.client.getSettings().toBuilder();
     } else {
-      return BigQueryWriteSettings.newBuilder()
-          .setCredentialsProvider(builder.credentialsProvider)
-          .setTransportChannelProvider(builder.channelProvider)
-          .setBackgroundExecutorProvider(builder.executorProvider)
-          .setEndpoint(builder.endpoint)
-          .build();
+      settingsBuilder =
+          new BigQueryWriteSettings.Builder()
+              .setTransportChannelProvider(
+                  BigQueryWriteSettings.defaultGrpcTransportProviderBuilder()
+                      .setChannelsPerCpu(1)
+                      .build())
+              .setCredentialsProvider(
+                  BigQueryWriteSettings.defaultCredentialsProviderBuilder().build())
+              .setBackgroundExecutorProvider(
+                  BigQueryWriteSettings.defaultExecutorProviderBuilder().build())
+              .setEndpoint(BigQueryWriteSettings.getDefaultEndpoint());
     }
+    if (builder.channelProvider != null) {
+      settingsBuilder.setTransportChannelProvider(builder.channelProvider);
+    }
+    if (builder.credentialsProvider != null) {
+      settingsBuilder.setCredentialsProvider(builder.credentialsProvider);
+    }
+    if (builder.executorProvider != null) {
+      settingsBuilder.setBackgroundExecutorProvider(builder.executorProvider);
+    }
+    if (builder.endpoint != null) {
```

Comment:
this is always triggered, endpoint is never null, it's default to non-null and never assigned as null, so it will always overwrite the one given in the client:
https://github.com/googleapis/java-bigquerystorage/blob/main/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java#L545
https://github.com/googleapis/java-bigquerystorage/blob/main/google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java#L597

Reply:
missed the later part of this file.